### PR TITLE
Chat Window: Load history, end to end working nostr chat, UI Improvements

### PR DESCRIPTION
**Pull request by abhay-raizada**
Make chat work, solvves https://github.com/lnflash/flash-mobile/issues/100

### CODE REVIEW

**Comment by abhay-raizada** [comment]
1. Chat screen now shows nostr users you've previously messaged:
image

2. Message screen now loads up past messages: 
image

3. UI enhancements in the message screen 
4. Made messages fuctional

**Comment by islandbitcoin** [comment]
Feedback: (see screenshots)

**Chat:**
- on Android, the "no messages here yet" is displaying mirror reversed. on iOS is it not reversed.
- not able to single-click on user to open up chat window, feels like I have to click twice, once to remove the keyboard and then again to open the window.
- I still cannot see the send message button on Android.
- after sending a message, under the message there is a "received message" label. That should be either much smaller compared to the actual note, or it should be hidden
- message are being sent successfully, but no message responses are being received in flash when I get a reply from another nostr client
- on IOS, the chat window gets pushed up to the top of the screen and I cannot see the chat history. everything feels a bit too large. 
- When pressing back and then going into the chat window again, all history is lost. 

**Search:**
- I created two flash users. When typing in the npub from one of those flash users into the search bar, I cannot find the user, I get no results. This might be because the user has not posted anything or finished setting up the profile?

**Pay:**
- pay button does not auto-fill in the lightning address of the recipient when going to the send screen. this should skip the initial send screen and already have the lightning address populated and only ask the user for the amount to send.

Screenshots:
IMAGE 2024-03-11 09:22:34
IMAGE 2024-03-11 09:23:12
IMAGE 2024-03-11 09:23:19
IMAGE 2024-03-11 09:23:24

File: app/config/galoy-instances.ts [hunk 1]
```diff
@@ -71,11 +71,11 @@ export const GALOY_INSTANCES: readonly GaloyInstance[] = [
...
     id: "Staging",
     name: "Staging",
-    graphqlUri: "https://api.development.flashapp.me:8080/graphql",
-    graphqlWsUri: "ws://ws.development.flashapp.me:4000/graphql",
-    authUrl: "https://api.development.flashapp.me:8080",
-    posUrl: "http://development.flashapp.me:3000",
-    lnAddressHostname: "development.flashapp.me:3000",
+    graphqlUri: "https://api.staging.flashapp.me/graphql",
+    graphqlWsUri: "ws://ws.staging.flashapp.me/graphql",
+    authUrl: "https://api.staging.flashapp.me:8080",
```
Comment (islandbitcoin) [inline app/config/galoy-instances.ts:76]: remove port 8080

File: app/hooks/use-nostr-profile.ts [hunk 2]
```diff
@@ -1,28 +1,44 @@
...
+  UnsignedEvent,
+  finalizeEvent,
+} from "nostr-tools"
 
 const useNostrProfile = () => {
   const KEYCHAIN_NOSTRCREDS_KEY = "nostr_creds_key"
   const [nostrSecretKey, setNostrSecretKey] = useState<string>("")
-  const relays = ["wss://relay.damus.io", "wss://relay.primal.net", "wss://purplepag.es"]
+  const [nostrPublicKey, setNostrPublicKey] = useState<string>("")
+  const relays = ["wss://relay.damus.io", "wss://relay.primal.net", "wss://nos.lol"]
```
Comment (islandbitcoin) [inline app/hooks/use-nostr-profile.ts:17]: definitely add 3 or 4 more relays:
wss://purplerelay.com/
wss://relay.snort.social/
wss://nostr.bitcoiner.social/
wss://nostr.oxtr.dev/
wss://relay.mostr.pub/
wss://purplerelay.com

**Review by islandbitcoin — CHANGES_REQUESTED** [review]

**Review by islandbitcoin — APPROVED** [review]
ACK

**Pull request merged**